### DISCUSSION TITLE
If the "Continuously active VPN" function is activated, Feeder does not recognise the network status

### DISCUSSION THREAD

**Discussion by OrkoGrayskull** [spacecowboy/Feeder#155] [General]
### Checklist

- [X] I have used the search function to see if someone else has already submitted the same bug report.
- [x] I will describe the problem with as much detail as possible.

### App version

2.6.11

### Where did you get the app from?

F-Droid

### Android version

14

### Device model

Pixel 6a

### Steps to reproduce

Enable VPN with the option "Continuously active VPN" under Android Settings.

### Expected behavior

If the "Continuously active VPN" function is activated, Feeder should recognise when the flight mode is switched on or no network connection is available.

### Actual behavior

If the option "Continuously active VPN" is activated for a VPN, Feeder tries to establish connections even if the device is in flight mode.

With NetGuard and RethinkDNS you can view the outgoing connection attempts of Feeder.

**Comment by spacecowboy** [comment]
What network traffic are we talking about?

It's not a bug per se since the VPN is still active I gather, so the system IS telling Feeder there is a network.

It's also not a problem (please clarify if it is). The requests just fail

**Comment by OrkoGrayskull** [comment]
> What network traffic are we talking about?

Feeder tries to fetch the RSS feed sources. 

> It's not a bug per se since the VPN is still active I gather, so the system IS telling Feeder there is a network.

There seem to be apps that perform an additional check and thus determine that there is no connection.

K-9 Mail and Element (Matrix) have similar problems:

- Thunderbird: https://github.com/thunderbird/thunderbird-android/issues/7408
- Elemental: https://github.com/element-hq/element-android/issues/8359

**Reply by spacecowboy** [reply]
I'd say this is a bug in Android if anything. The whole point is that apps shouldn't care (or even be aware of) if you are using a VPN.

Android is misbehaving by not reporting the network as offline when you are using this kind of VPN.

**Reply by spacecowboy** [reply]
Again. It's not a problem. The requests fail, no big deal.

You would only be aware of it if you read the logs, which 99% of users won't.

**Reply by ignoramous** [reply]
*rdns dev here*

> Android is misbehaving by not reporting the network as offline when you are using this kind of VPN.

Can you please point out how Feeder determines if the active network has connectivity or not? 

> Again. It's not a problem. The requests fail, no big deal.

Not sure about Feeder, but the Elemental Android app (linked above) drains battery due to a large number of ceaseless retries.

**Reply by spacecowboy** [reply]
Feeder uses the official workmanager library: https://developer.android.com/topic/libraries/architecture/workmanager/

which handles job scheduling.

The constraints are defined here: 
https://github.com/spacecowboy/Feeder/blob/a0c4a8b860d8d2f37f4d7248e31263cb0eb0a144/app/src/main/java/com/nononsenseapps/feeder/archmodel/SettingsStore.kt#L452

either unmetered or connected.

so this is a bug in work manager if anything.

Battery drain is constrained by your sync schedule.

**Reply by ulairii** [reply]
Would it be possible to add a customized, app-level server reachability check before trying synchronization? This could help prevent redundant sync attempts when the app is effectively blocked by the VPN.

Here are some references that may help:

[StackOverflow: Server reachability check](https://stackoverflow.com/a/4009133/18154447)

[DAVx⁵ solution for handling VPN-related sync issues](https://github.com/bitfireAT/davx5-ose/commit/318939e970297e9d491b82cbe5238adea82fd969)

Thank you!

**Reply by ignoramous** [reply]
> so this is a bug in work manager if anything

In the upcoming Rethink version, we are implementing a suggestion made by GrapheneOS which may be hints the WorkManager to stop retries: https://github.com/celzero/rethink-app/issues/1157#issuecomment-2453116167
